docs/source/API/core/policies/MDRangePolicy.rst

```diff
@@ -86,7 +86,55 @@ Constructors
 
     * Provide a start and end index as well as the tiling dimensions. The length of the lists must match the rank of the policy.
 
+CTAD Constructors (since 4.3):
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: cppkokkos
+
+   DefaultExecutionSpace des;
+   SomeExecutionSpace ses; // different from DefaultExecutionSpace
+   int64_t i;
+
+   // Deduces to MDRangePolicy<Rank<5>>
+   MDRangePolicy pl0({1, 2, 3, 4, 5}, {1, 2, 3, 4, 5});
+   MDRangePolicy pl1({1, 2, 3, 4, 5}, {1, 2, 3, 4, 5}, { i });
+   MDRangePolicy pl2(des, {1, 2, 3, 4, 5}, {1, 2, 3, 4, 5});
+   MDRangePolicy pl3(des, {1, 2, 3, 4, 5}, {1, 2, 3, 4, 5}, { i });
+
+   // Deduces to MDRangePolicy<SomeExecutionSpace, Rank<5>>
+   MDRangePolicy pl4(ses, {1, 2, 3, 4, 5}, {1, 2, 3, 4, 5});
+   MDRangePolicy pl5(ses, {1, 2, 3, 4, 5}, {1, 2, 3, 4, 5}, { i });
+
+   int cbegin[3];
+   int cend[3];
+   int64_t ctiling[2];
```

Comment:
```suggestion
   int ctiling[3];
```
?

Reply:
The change in type exhibits the flexibility those constructors have.

Reply:
I need
```
diff --git a/example/build_cmake_in_tree/cmake_example.cpp b/example/build_cmake_in_tree/cmake_example.cpp
index 8b06caee8..dcfbc5ae8 100644
--- a/example/build_cmake_in_tree/cmake_example.cpp
+++ b/example/build_cmake_in_tree/cmake_example.cpp
@@ -53,7 +53,7 @@ int main(int argc, char* argv[]) {
 
   Kokkos::Array<int, 2> abegin;
   Kokkos::Array<int, 2> aend;
-  Kokkos::Array<int64_t, 2> atiling;
+  Kokkos::Array<int, 2> atiling;
 
   // Deduces to Kokkos::MDRangePolicy<Rank<2>>
   Kokkos::MDRangePolicy pa0(abegin, and);
```
for the whole example to compile. Error look like
```
/tmp/kokkos_new/example/build_cmake_in_tree/cmake_example.cpp:60:23: error: no viable constructor or deduction guide for deduction of template arguments of 'MDRangePolicy'
   60 | Kokkos::MDRangePolicy pa1(abegin, aend, atiling);
      |                       ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:421:1: note: candidate template ignored: deduced conflicting types for parameter 'T' ('int' vs. 'long long')
  421 | MDRangePolicy(Array<T, N> const&, Array<T, N> const&, Array<T, NT> const&)
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:425:1: note: candidate template ignored: deduced conflicting types for parameter 'T' ('int' vs. 'long long')
  425 | MDRangePolicy(DefaultExecutionSpace const&, Array<T, N> const&,
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:436:1: note: candidate template ignored: deduced conflicting types for parameter 'T' ('int' vs. 'long long')
  436 | MDRangePolicy(ES const&, Array<T, N> const&, Array<T, N> const&)
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:395:1: note: candidate template ignored: could not match 'LT[N]' against 'Kokkos::Array<int, 2>'
  395 | MDRangePolicy(const LT (&)[N], const UT (&)[N], const TT (&)[TN])
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:399:1: note: candidate template ignored: could not match 'LT[N]' against 'Kokkos::Array<int, 2>'
  399 | MDRangePolicy(DefaultExecutionSpace const&, const LT (&)[N], const UT (&)[N])
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:409:1: note: candidate template ignored: could not match 'LT[N]' against 'Kokkos::Array<int, 2>'
  409 | MDRangePolicy(ES const&, const LT (&)[N], const UT (&)[N])
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:392:1: note: candidate function template not viable: requires 2 arguments, but 3 were provided
  392 | MDRangePolicy(const LT (&)[N], const UT (&)[N])->MDRangePolicy<Rank<N>>;
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:418:1: note: candidate function template not viable: requires 2 arguments, but 3 were provided
  418 | MDRangePolicy(Array<T, N> const&, Array<T, N> const&)->MDRangePolicy<Rank<N>>;
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:403:1: note: candidate function template not viable: requires 4 arguments, but 3 were provided
  403 | MDRangePolicy(DefaultExecutionSpace const&, const LT (&)[N], const UT (&)[N],
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  404 |               const TT (&)[TN])
      |               ~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:414:1: note: candidate function template not viable: requires 4 arguments, but 3 were provided
  414 | MDRangePolicy(ES const&, const LT (&)[N], const UT (&)[N], const TT (&)[TN])
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:430:1: note: candidate function template not viable: requires 4 arguments, but 3 were provided
  430 | MDRangePolicy(DefaultExecutionSpace const&, Array<T, N> const&,
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  431 |               Array<T, N> const&, Array<T, NT> const&)
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:441:1: note: candidate function template not viable: requires 4 arguments, but 3 were provided
  441 | MDRangePolicy(ES const&, Array<T, N> const&, Array<T, N> const&,
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  442 |               Array<T, NT> const&)
      |               ~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:158:8: note: candidate function template not viable: requires 1 argument, but 3 were provided
  158 | struct MDRangePolicy;
      |        ^~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:158:8: note: candidate function template not viable: requires 0 arguments, but 3 were provided
/tmp/kokkos_new/example/build_cmake_in_tree/cmake_example.cpp:62:23: error: no viable constructor or deduction guide for deduction of template arguments of 'MDRangePolicy'
   62 | Kokkos::MDRangePolicy pa3(des, abegin, aend, atiling);
      |                       ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:430:1: note: candidate template ignored: deduced conflicting types for parameter 'T' ('int' vs. 'long long')
  430 | MDRangePolicy(DefaultExecutionSpace const&, Array<T, N> const&,
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:441:1: note: candidate template ignored: deduced conflicting types for parameter 'T' ('int' vs. 'long long')
  441 | MDRangePolicy(ES const&, Array<T, N> const&, Array<T, N> const&,
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:403:1: note: candidate template ignored: could not match 'LT[N]' against 'Kokkos::Array<int, 2>'
  403 | MDRangePolicy(DefaultExecutionSpace const&, const LT (&)[N], const UT (&)[N],
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:414:1: note: candidate template ignored: could not match 'LT[N]' against 'Kokkos::Array<int, 2>'
  414 | MDRangePolicy(ES const&, const LT (&)[N], const UT (&)[N], const TT (&)[TN])
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:395:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  395 | MDRangePolicy(const LT (&)[N], const UT (&)[N], const TT (&)[TN])
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:399:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  399 | MDRangePolicy(DefaultExecutionSpace const&, const LT (&)[N], const UT (&)[N])
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:409:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  409 | MDRangePolicy(ES const&, const LT (&)[N], const UT (&)[N])
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:421:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  421 | MDRangePolicy(Array<T, N> const&, Array<T, N> const&, Array<T, NT> const&)
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:425:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  425 | MDRangePolicy(DefaultExecutionSpace const&, Array<T, N> const&,
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  426 |               Array<T, N> const&)
      |               ~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:436:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  436 | MDRangePolicy(ES const&, Array<T, N> const&, Array<T, N> const&)
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:392:1: note: candidate function template not viable: requires 2 arguments, but 4 were provided
  392 | MDRangePolicy(const LT (&)[N], const UT (&)[N])->MDRangePolicy<Rank<N>>;
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:418:1: note: candidate function template not viable: requires 2 arguments, but 4 were provided
  418 | MDRangePolicy(Array<T, N> const&, Array<T, N> const&)->MDRangePolicy<Rank<N>>;
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:158:8: note: candidate function template not viable: requires 1 argument, but 4 were provided
  158 | struct MDRangePolicy;
      |        ^~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:158:8: note: candidate function template not viable: requires 0 arguments, but 4 were provided
/tmp/kokkos_new/example/build_cmake_in_tree/cmake_example.cpp:66:23: error: no viable constructor or deduction guide for deduction of template arguments of 'MDRangePolicy'
   66 | Kokkos::MDRangePolicy pa5(ses, abegin, aend, atiling);
      |                       ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:430:1: note: candidate template ignored: deduced conflicting types for parameter 'T' ('int' vs. 'long long')
  430 | MDRangePolicy(DefaultExecutionSpace const&, Array<T, N> const&,
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:441:1: note: candidate template ignored: deduced conflicting types for parameter 'T' ('int' vs. 'long long')
  441 | MDRangePolicy(ES const&, Array<T, N> const&, Array<T, N> const&,
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:403:1: note: candidate template ignored: could not match 'LT[N]' against 'Kokkos::Array<int, 2>'
  403 | MDRangePolicy(DefaultExecutionSpace const&, const LT (&)[N], const UT (&)[N],
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:414:1: note: candidate template ignored: could not match 'LT[N]' against 'Kokkos::Array<int, 2>'
  414 | MDRangePolicy(ES const&, const LT (&)[N], const UT (&)[N], const TT (&)[TN])
      | ^
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:395:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  395 | MDRangePolicy(const LT (&)[N], const UT (&)[N], const TT (&)[TN])
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:399:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  399 | MDRangePolicy(DefaultExecutionSpace const&, const LT (&)[N], const UT (&)[N])
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:409:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  409 | MDRangePolicy(ES const&, const LT (&)[N], const UT (&)[N])
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:421:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  421 | MDRangePolicy(Array<T, N> const&, Array<T, N> const&, Array<T, NT> const&)
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:425:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  425 | MDRangePolicy(DefaultExecutionSpace const&, Array<T, N> const&,
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  426 |               Array<T, N> const&)
      |               ~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:436:1: note: candidate function template not viable: requires 3 arguments, but 4 were provided
  436 | MDRangePolicy(ES const&, Array<T, N> const&, Array<T, N> const&)
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:392:1: note: candidate function template not viable: requires 2 arguments, but 4 were provided
  392 | MDRangePolicy(const LT (&)[N], const UT (&)[N])->MDRangePolicy<Rank<N>>;
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:418:1: note: candidate function template not viable: requires 2 arguments, but 4 were provided
  418 | MDRangePolicy(Array<T, N> const&, Array<T, N> const&)->MDRangePolicy<Rank<N>>;
      | ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:158:8: note: candidate function template not viable: requires 1 argument, but 4 were provided
  158 | struct MDRangePolicy;
      |        ^~~~~~~~~~~~~
/tmp/kokkos_new/core/src/KokkosExp_MDRangePolicy.hpp:158:8: note: candidate function template not viable: requires 0 arguments, but 4 were provided
3 errors generated.
```

Reply:
Fixed (I didn't realize that for `Kokkos::Array` they had to be the same type).